py3status/py3.py

```diff
@@ -679,3 +740,67 @@ def threshold_get_color(self, value, name=None):
         setattr(self._py3status_module, color_name, color)
 
         return color
+
+    def _get_url(self, url):
+        """
+        Get the HTTPResponse for the HTTP GET request to `url`.
+
+        Raises `Py3.URLError` on Connection Errors.
+        Raises `Py3.ClientError` on HTTP 4xx Errors.
+        Raises `Py3.ServerError` on HTTP 5xx Errors.
+        """
+        try:
+            response = HTTPResponse(urlopen(url))
```

Comment:
Why not just pass the url to HTTPResponse and let it decide what to do?  that way it's all self contained.